runelite-client/src/main/java/net/runelite/client/plugins/hunter/HunterPlugin.java

```diff
@@ -377,6 +390,89 @@ else if (containsBoulder) // For traps like deadfalls. This is different because
 		lastTickLocalPlayerLocation = client.getLocalPlayer().getWorldLocation();
 	}
 
+	@Subscribe
+	public void onInteractingChanged(InteractingChanged interactingChanged)
+	{
+		Actor target = interactingChanged.getTarget();
+		Actor source = interactingChanged.getSource();
+
+		mapActorToChinTarget(target, source);
+	}
+
+	/**
+	 *	 				DEAD TRAP MECHANICS
+	 * 	 if two hunter traps target the same Actor, then the trap
+	 * 	 which did not catch the target will get stuck in a "dead" state.
+	 * 	 It will no longer be able to re-target another actor.
+	 * 	 Therefore, the user can remove these traps and re-place them.
+	 *
+	 *	 The code below keeps track of which traps interacted with every
+	 *	 chinchompa. If any traps are left over in this list when
+	 *	 the chinchompa is caught, the trap enters a dead state.
+	 */
+	private void mapActorToChinTarget(Actor target, Actor source)
+	{
+		if (target != null && source != null)
+		{
+			// May be a trap (trap source name is null)
+			if (source.getName() == null)
+			{
+				traps.forEach((worldPoint, trap) ->
+				{
+					// Check if source is a trap in the list of traps the player put down
+					if (source.getWorldLocation().distanceTo(worldPoint) == 0)
+					{
+						// Check if chin target entry doesnt exist yet, otherwise add it
+						if (chinTargets.get(target) == null)
+						{
+							// add new entry with new list of traps for the chin
+							chinTargets.put(target, new ArrayList<>());
+							chinTargets.get(target).add(trap);
+						}
+						else
+						{
+							if (!chinTargets.get(target).contains(trap)) chinTargets.get(target).add(trap);
+						}
+					}
+				});
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameObjectDespawned(GameObjectDespawned gameObjectDespawned)
+	{
+		GameObject object = gameObjectDespawned.getGameObject();
+		HunterTrap trap = traps.get(object.getWorldLocation());
+		if (traps.containsKey(object.getWorldLocation()))
+		{
+			chinTargets.forEach((key, value) -> value.remove(trap));
+		}
+	}
+
+	@Subscribe
+	public void onNpcDespawned(NpcDespawned npcDespawned)
+	{
+		NPC npc = npcDespawned.getNpc();
+		if (chinIds.contains(npc.getId()))
+		{
+			// if chin has 2+ targets it means the traps leftover will turn dead
+			if (chinTargets.get(npcDespawned.getActor()).size() >= 2)
```

Comment:
Store `chinTargets.get()` to a variable so it does not need to be fetched multiple times within this block.